### DISCUSSION TITLE
Implementing a filter for recent transactions

### DISCUSSION THREAD

**Discussion by stark-3k** [mempool/mempool#5217] [Q&A] [answered]
Hi, I am working on implementing a filter for recent transactions(related to #3960), just need some clarifications on how to go about it.
There are a few ways to implement a filter:
1. Let the user input the minimum amount and request the transactions from server/backend
2. Let user choose from an option of minimum amounts(e.g. - 0.1 BTC, 1 BTC, 10 BTC) and send all transactions corresponding to these filters via ws so the user can switch fast
3. Send a bunch of recent transactions to the frontend and add filters in the frontend

Also, is creating a new `/transactions` route to display recent transactions a better idea?

I have created a naive implementation by filtering the transactions in the frontend for reference:
![mempool](https://github.com/mempool/mempool/assets/62797354/990a56dd-a722-4806-8b7c-d917e8f731c3)

**Comment by softsimon** [comment, marked as answer]
1. Using a frontend filter for incoming transactions might be useful if the user want to see only a certain type of transaction to show up. But it's a little bit weird feature as it only applies to new transactions that come in and disappear from the list fast
2. A proper feature that covers all use cases would be a new separate page at `/transactions` that list all mempool-transactions paginated and added live using the already existing WebSocket API. Then add a bunch of filters on top of the page where the user can find by any type of size etc etc, even by date since we use the FirstSeen date when the transaction arrived. So the filters will be handled on the backend, since we shouldn't send the full mempool to the frontend (too much data).

**Reply by stark-3k** [reply]
Sounds good. Will work on it.
